cpp/demo/poisson_matrix_free/CMakeLists.txt

```diff
@@ -9,6 +9,7 @@ project(${PROJECT_NAME} LANGUAGES C CXX)
 
 # Set C++20 standard
 set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_C_STANDARD 17)
```

Comment:
Any reason for C++17 here? Could cause problems when including header files.

I think the CMake files are generated, so changes should be make in the generating script.

Reply:
Not the first to misread this - it's C17, not C++17.